### ISSUE TITLE
Application hangs

### ISSUE THREAD

**Issue by JonnyTech** [celzero/rethink-app#1629]
On my Android 13 device, I start the Rethink app and leave it running. The persistent notification remains in the task bar. Later when I want to stop it, I try to click the `STOP` label in the notification or the stop button in the app but neither respond. The only solution is to force quit the app, then open it again, then I can click the stop button. Is this a known bug and is there a fix available?

**Comment by ignoramous** [comment]
I've seen this happens at times, and it is annoying, and has multiple reasons. We continue to improve the way we stop of the VPN tunnel and the app's [*network engine*](https://github.com/celzero/firestack). In `v055o` (the next version due in a week or so) we've put in a fixed heartbeat-like liveliness timeout ([code](https://github.com/celzero/firestack/blob/f04fe1433cfdbce69b417bf0ad727fda7f54cd83/tunnel/tunnel.go#L181-L222)) to forcefully stop the network engine after 15s of inactivity.

**Comment by JonnyTech** [comment]
Thank you for your feedback.

> to forcefully stop the network engine after 15s of inactivity.

I want to keep the engine alive until I decide to stop it myself. I am not a heavy user so am now worried that the next version will disconnect often.

The issue is with the GUI app interface, that is what locks up, not the engine. Is there any way to revive that, instead of force closing then restarting it?

**Comment by ignoramous** [comment]
> I want to keep the engine alive until I decide to stop it myself. I am not a heavy user so am now worried that the next version will disconnect often.

Of course, this timeout doesn't work the way you think it does. I specifically mentioned "heartbeat" and "liveliness" for a reason (:

> The issue is with the GUI app interface, that is what locks up, not the engine. Is there any way to revive that, instead of force closing then restarting it?

If it is the UI that's locked up, Android should ANR (activity not responding) crash it in 5s to 10s. If not, something else is (in our experience, it is the *network engine*, which is a complicated piece of code, that sometimes takes too long to... *close*).

**Comment by JonnyTech** [comment]
Great - awaiting `v055o` to drop in FDroid then I shall test and report back :+1:

**Comment by ignoramous** [comment]
v055o is out: https://github.com/celzero/rethink-app/releases/tag/v0.5.5o

May make it on F-Droid by next week.

**Comment by ignoramous** [comment]
Should have been fixed in `v055u` even if not in `v055u`, which is the latest.

(closing, but feel free to ask to reopen if you encounter this issue still on current version, which, as of today, is [`v055u`](https://github.com/celzero/rethink-app/releases/tag/v0.5.5u) on GitHub/Website/PlayStore and `v055t` on F-Droid).
